catalog/core/catalog-core-subscriptionstore/src/main/java/org/codice/ddf/catalog/subscriptionstore/internal/SubscriptionStoreException.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.subscriptionstore.internal;
+
+/**
+ * Exception specific to the subscription store.
+ */
+public class SubscriptionStoreException extends RuntimeException {
```

Comment:
I modeled the contract similar to how the CSW subscription endpoint already behaved. `null` was a valid return type, which usually got converted to a `404`, so I didn't see the need to force the API consumer to handle the exception. 

Technically, it should "just work" if you (1) pass correct data and (2) account for a `null` return type. I consider the exception as either a consumer implementation error or an internal persistence issue.